public/static/js/task.js

```diff
@@ -586,7 +586,25 @@ mciModule.controller('TaskLogCtrl', ['$scope', '$timeout', '$http', '$location',
     return '[' + timestamp + '] '
   }
 
+  var isUnfinished = function(status) {
+    switch (status) {
+      case "undispatched":
+        return true;
+      case "started":
+        return true;
+      case "dispatched":
+        return true;
+      case "unstarted":
+        return true;
+      default:
+        return false;
+    }
+  }
+
   $scope.getLogs = function() {
+    if (!isUnfinished($scope.task.status)) {
```

Comment:
It would be clearer to define this function as `isFinished` so you don't have to invert a negative.